jdbccore/src/main/java/it/eng/knowage/tools/utils/DatabaseUtils.java

```diff
@@ -26,12 +26,12 @@ public static Object timestampFormatter(Object value) {
 				Timestamp timestamp = (Timestamp) value;
 				return timestamp.getTime();
 			} else {
-				SimpleDateFormat timestampFormatter = new SimpleDateFormat(TIMESTAMP_FORMAT);
+				DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(TIMESTAMP_FORMAT);
+				Instant instant = dateTimeFormatter.parse((String) value, Instant::from);
```

Comment:
This simple code is throwing exception:

```
String value = "2023-09-11 15:05:59.6";
DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(TIMESTAMP_FORMAT);
Instant instant = dateTimeFormatter.parse(value, Instant::from);
```
Exception:

Exception in thread "main" java.time.format.DateTimeParseException: Text '2023-09-11 15:05:59.6' could not be parsed: Unable to obtain Instant from TemporalAccessor: {},ISO resolved to 2023-09-11T15:05:59.600 of type java.time.format.Parsed
	at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:1920)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1855)
	at .....
Caused by: java.time.DateTimeException: Unable to obtain Instant from TemporalAccessor: {},ISO resolved to 2023-09-11T15:05:59.600 of type java.time.format.Parsed
	at java.time.Instant.from(Instant.java:378)
	at java.time.format.Parsed.query(Parsed.java:226)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
	... 1 more
Caused by: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: InstantSeconds
	at java.time.format.Parsed.getLong(Parsed.java:203)
	at java.time.Instant.from(Instant.java:373)
	... 3 more

Please check

Reply:
Replaced Instant::from, it seems it isn't compatible with our JDK.

Reply:
> Replaced Instant::from, it seems it isn't compatible with our JDK.

Do you remeber more about this? Because I would say you are on a JDK 8 like:

https://adoptopenjdk.net/releases.html?variant=openjdk8&jvmVariant=hotspot

With a source level of 8 as:

https://github.com/KnowageLabs/Knowage-Server/blob/master/knowage-ce-parent/pom.xml#L69

The lambda should be ok.

Because during the discussion about this with Davide, I proposed to replace SimpeDateFormat with DateTimeFormatter which is slower but synchronized and it should avoid the creation of new methods and synchronized blocks.

Reply:
Hello @kerny3d , I tried it using some dummy test cases, but it didn't work, and I changed this entire code so that we are using the first proposed solution from a pdf file, synchronized block. The problem with date formatting classes is that I'm not sure how to change it in some files, what format to expect etc. so I think the solution with synchronized could be better if we don't want to change the logic.

Reply:
Hello @kerny3d, should I change it to use DateTimeFormatter? The problem is that I'm not sure what formats to expect in some files and replacing the SimpeDateFormat is not so easy (for example we use classes that have fields as Object which could be anything I guess).